src/internal/operations/pathtransformer/restore_path_transformer.go

```diff
@@ -0,0 +1,180 @@
+package pathtransformer
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/internal/version"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+func locationRef(
+	ent *details.Entry,
+	repoRef path.Path,
+	backupVersion int,
+) (*path.Builder, error) {
+	loc := ent.LocationRef
+
+	// At this backup version all data types should populate LocationRef.
+	if len(loc) > 0 || backupVersion >= version.OneDrive7LocationRef {
+		return path.Builder{}.SplitUnescapeAppend(loc)
+	}
+
+	// We could get an empty LocationRef either because it wasn't populated or it
+	// was in the root of the data type.
+	elems := repoRef.Folders()
+
+	if ent.OneDrive != nil || ent.SharePoint != nil {
+		dp, err := path.ToDrivePath(repoRef)
+		if err != nil {
+			return nil, clues.Wrap(err, "fallback for LocationRef")
+		}
+
+		elems = append([]string{dp.Root}, dp.Folders...)
+	}
+
+	return path.Builder{}.Append(elems...), nil
+}
+
+func basicPathMerge(repoRef path.Path, locRef *path.Builder) (path.Path, error) {
```

Comment:
hmm, `merge` feels like the wrong name here.  It implies that you're merging the two paths into one, which isn't exactly what's happening.  What about:
```suggestion
func basicLocationPath(repoRef path.Path, locRef *path.Builder) (path.Path, error) {
```